### PR TITLE
First revision password grant

### PR DESCRIPTION
**Pull request by FlinkbaumFAU**

### CODE REVIEW

**Comment by skriesch** [comment]
I have tried to build it and receive:
> openid-doctor@0.0.1 prebuild
> rimraf dist
/tmp/prebuild6573395847.sh: Zeile 2: rimraf: Kommando nicht gefunden.

Did you integrate rimraf into the package config?

**Comment by skriesch** [comment]
> I have tried to build it and receive:
> 
> > openid-doctor@0.0.1 prebuild
> > rimraf dist
> > /tmp/prebuild6573395847.sh: Zeile 2: rimraf: Kommando nicht gefunden.
> 
> Did you integrate rimraf into the package config?

It seems, that this issue does not come from you. Sorry!

**Comment by skriesch** [comment]
I do not see any issues or anything else after the installation of rimraf. The frontend configuration works also.

**Review by skriesch — APPROVED** [review]

**Comment by skriesch** [comment]
I found, that the decoding part has been only replaced to another place. The rest looks good for me.

File: src/flows/flows.service.ts [hunk 1]
```diff
@@ -77,56 +125,97 @@ export class FlowsService {
...
-      .decodeToken(receivedTokenString)
-      .then(async ([header, payload]) => {
-        const validationResult = await this.tokenService
-          .validateTokenSignature(issuer_s, receivedTokenString, true, '', '')
-          .then(async ([isValid, message]) => {
-            if (isValid) {
-              await this.utilsService.writeOutput(
-                header + '\n' + payload + '\n' + message,
-              );
-              this.protocolService.extendedLogSuccess("Token decoded");
```
[the remark this answers is not in the document]
Comment (FlinkbaumFAU) [reply src/flows/flows.service.ts:100]: Yes it was a task in one of the last sprints. It was implemented in token.service.
This function just uses this implementation and checks if an error occured

**Pull request merged**